src/main/kotlin/com/example/passmanager/web/controller/PassOwnerController.kt

```diff
@@ -30,41 +32,41 @@ internal class PassOwnerController(
 ) {
 
     @GetMapping("/{id}/distributions")
-    fun calculatePriceDistributions(@PathVariable id: String): ResponseEntity<List<PriceDistribution>> {
-        val priceDistributions = passOwnerStatisticsService.calculatePriceDistributions(id)
-        return ResponseEntity.ok(priceDistributions)
+    fun calculatePriceDistributions(@PathVariable id: String): ResponseEntity<Flux<PriceDistribution>> {
+        return ResponseEntity.ok(passOwnerStatisticsService.calculatePriceDistributions(id))
     }
 
     @GetMapping("/{id}/spent")
     fun calculateSpentAfterDate(
         @RequestParam afterDate: LocalDate,
         @PathVariable("id") ownerId: String,
-    ): ResponseEntity<SpentAfterDateDto> {
-        val totalSpent = passOwnerStatisticsService.calculateSpentAfterDate(afterDate, ownerId)
-        return ResponseEntity.ok(SpentAfterDateDto(ownerId, afterDate, totalSpent))
+    ): Mono<ResponseEntity<SpentAfterDateDto>> {
+        return passOwnerStatisticsService.calculateSpentAfterDate(afterDate, ownerId)
+            .map { spentAfterDate -> SpentAfterDateDto(ownerId, afterDate, spentAfterDate) }
+            .map { ResponseEntity.ok(it) }
     }
 
     @PostMapping
-    fun create(@Valid @RequestBody dto: PassOwnerDto): ResponseEntity<PassOwnerDto> {
-        val created = passOwnerService.create(dto.toEntity())
-        return ResponseEntity.status(HttpStatus.CREATED).body(created.toDto())
+    fun create(@Valid @RequestBody dto: PassOwnerDto): Mono<ResponseEntity<PassOwnerDto>> {
+        return passOwnerService.create(dto.toEntity())
+            .map { it.toDto() }
+            .map { ResponseEntity.status(HttpStatus.CREATED).body(it) }
     }
 
     @PatchMapping("/{id}")
     fun partialUpdate(
         @Valid @RequestBody dto: PassOwnerDto,
         @PathVariable("id") ownerId: String,
-    ): ResponseEntity<PassOwnerDto> {
-        return passOwnerService.findById(ownerId)?.let { passOwnerInDb ->
-            val partiallyUpdated = passOwnerInDb.partialUpdate(dto)
-            val updated = passOwnerService.update(partiallyUpdated)
-            return ResponseEntity.ok(updated.toDto())
-        } ?: ResponseEntity.notFound().build()
+    ): Mono<ResponseEntity<PassOwnerDto>> {
+        return passOwnerService.getById(ownerId)
+            .map { passOwner -> passOwner.partialUpdate(dto) }
+            .flatMap { partiallyUpdated -> passOwnerService.update(partiallyUpdated) }
```

Comment:
the controller function looks too complex. Should we move all the chain invocation into a service?